lib/galaxy/webapps/galaxy/api/visualizations.py

```diff
@@ -239,175 +234,44 @@ def set_slug(
-
-        # keys listed here don't error when attempting to set, but fail silently
-        #   this allows PUT'ing an entire model back to the server without attribute errors on uneditable attrs
-        valid_but_uneditable_keys = (
-            "id",
-            "model_class",
-            # TODO: fill out when we create to_dict, get_dict, whatevs
-        )
-        # TODO: importable
-        ValidationError = exceptions.RequestParameterInvalidException
-
-        validated_payload = {}
-        for key, val in payload.items():
-            # TODO: validate types in VALID_TYPES/registry names at the mixin/model level?
-            if key == "type":
-                if not isinstance(val, str):
-                    raise ValidationError(f"{key} must be a string or unicode: {str(type(val))}")
-                val = util.sanitize_html.sanitize_html(val)
-            elif key == "config":
-                if not isinstance(val, dict):
-                    raise ValidationError(f"{key} must be a dictionary: {str(type(val))}")
-            elif key == "annotation":
-                if not isinstance(val, str):
-                    raise ValidationError(f"{key} must be a string or unicode: {str(type(val))}")
-                val = util.sanitize_html.sanitize_html(val)
-            elif key == "deleted":
-                if not isinstance(val, bool):
-                    raise ValidationError(f"{key} must be a bool: {str(type(val))}")
-
-            # these are keys that actually only be *updated* at the revision level and not here
-            #   (they are still valid for create, tho)
-            elif key == "title":
-                if not isinstance(val, str):
-                    raise ValidationError(f"{key} must be a string or unicode: {str(type(val))}")
-                val = util.sanitize_html.sanitize_html(val)
-            elif key == "slug":
-                if not isinstance(val, str):
-                    raise ValidationError(f"{key} must be a string: {str(type(val))}")
-                val = util.sanitize_html.sanitize_html(val)
-            elif key == "dbkey":
-                if not isinstance(val, str):
-                    raise ValidationError(f"{key} must be a string or unicode: {str(type(val))}")
-                val = util.sanitize_html.sanitize_html(val)
-
-            elif key not in valid_but_uneditable_keys:
-                continue
-                # raise AttributeError( 'unknown key: %s' %( str( key ) ) )
-
-            validated_payload[key] = val
-        return validated_payload
+    @router.put(
+        "/api/visualizations/{id}",
+        summary="Update a visualization.",
+    )
+    def update(
+        self,
+        id: VisualizationIdPathParam,
+        payload: VisualizationUpdatePayload = Body(...),
+        trans: ProvidesUserContext = DependsOnTrans,
+    ) -> Union[VisualizationUpdateResponse, None]:
```

Comment:
It is similar, but we use Optional for consistency.
```suggestion
    ) -> Optional[VisualizationUpdateResponse]:
```